src/pages/SitePermissionPage.qml

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+**
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Page {
+    id: page
+
+    property string title
+    property string url
+
+    property var _permissions: {
+        "geolocation": PermissionManager.Prompt
+    }
+
+    signal requestFinished()
+
+    function getCurrentIndex(type) {
```

Comment:
The title `getCurrentIndex` causes some confusion for me, because I'm expecting it to get the index from the combobox. Perhaps this is a matter of taste, but for me something like `convertTypeToIndex` would be clearer.